pkg/controller/operators/olm/apiservices.go

```diff
@@ -921,3 +978,53 @@ func (a *Operator) isAPIServiceAdoptable(target *v1alpha1.ClusterServiceVersion,
 	adoptable = ownerutil.AdoptableLabels(apiService.GetLabels(), true, owners...)
 	return
 }
+
+// deleteLegacyAPIServiceResources deletes resources that were created by OLM for an APIService that used the old naming convention.
+func (a *Operator) deleteLegacyAPIServiceResources(namespace string, desc v1alpha1.APIServiceDescription) error {
+	apiServiceName := fmt.Sprintf("%s.%s", desc.Version, desc.Group)
+
+	// Attempt to delete the legacy service
+	err := a.opClient.DeleteService(namespace, legacyAPIServiceNameToServiceName(apiServiceName), &metav1.DeleteOptions{})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
```

Comment:
Could we not try to go on deleting the other resources, and return an aggregated error, instead of returning an error immediately and letting the other resources hang around?

Reply:
We could - but these errors are only returned if OLM encounters an error other than IsNotFound - likely a connection error in which case all other requests would fail as well.

Additionally, the CSV should continue to be reconciled, so the install should eventually succeed.